### PR TITLE
Add support for hyphenation to wrapped text

### PR DESCRIPTION
**Pull request by pepijndevos**
#### Description

This PR adds hyphenation to the word wrapping functionality. Example:

image

This needs a `poetry add pyphen` or similar but these things tend to create conflicts so I'd save that until all the tests and docs and API and bikeshedding is done.

#### Checklist

- [x] feature/fix implemented
- [x] code formatting ok (`black` and `ruff check .`)
- [x] `mypy` returns no error
- [ ] tests added/updated and `pytest` succeeds
- [x] add Pyphen dependency
- [x] make configurable
- [ ] documentation added/updated
    - [x] command docstring and option/argument `help`
    - [ ] README.md updated (Feature Overview)
    - [x] CHANGELOG.md updated
    - [ ] added new command to `reference.rst`
    - [ ] RTD doc updated and building with no error (`make clean && make html` in `docs/`)

### CODE REVIEW

**Review by abey79 — CHANGES_REQUESTED** [review]
Great contribution, thanks!

You will have to add the dependency to `pyproject.toml` for the CI to have a chance to succeed. 

Also, it would be nice to add the corresponding parameters to the `text` command in `vpype_cli`, and update all the docstrings.

File: vpype/text.py [hunk 1]
```diff
@@ -195,6 +207,7 @@ def text_block(
     align: str = "left",
     line_spacing: float = 1,
     justify=False,
+    hyphenate="en",
```
[the remark this answers is not in the document]
Comment (pepijndevos) [reply vpype/text.py:210]: I would think of doing `--hyphenate en` and make it also `str | None` maybe?

[the remark this answers is not in the document]
Comment (abey79) [reply vpype/text.py:210]: Yeah I guess that would work as well.

**Comment by pepijndevos** [comment]
Alright, I think that should be most of the relevant changes?

File: vpype_cli/text.py [hunk 2]
```diff
@@ -53,7 +60,8 @@ def text(
 
     In wrap mode, the text start at (0, 0) and expends left until it reach the specified width.
     The `--align` option controls how the text is laid out within the column and behaves as
-    typically expected.
+    typically expected. The `--hyphenate` options enables hyphenation using the provided
+    language code.
```
Comment (abey79) [inline vpype_cli/text.py:64]: Same here.

File: vpype/text.py [hunk 3]
```diff
@@ -210,6 +227,7 @@ def text_block(
             left alignment)
         line_spacing: line spacing (default: 1.0)
         justify: should the text be justified (default: False)
+        hyphenate: wrapped text is hyphenated with the given language (default: None)
```
Comment (abey79) [inline vpype/text.py:230]: It would be great to indicate what format(s) are acceptable for the language, and provide a couple of examples.

**Review by abey79 — APPROVED** [review]
Looks good! Just a small comment on the docs to point at the right format to specify the language.

**Comment by pepijndevos** [comment]
Something like this? Also caught a leftover print statement

**Comment by abey79** [comment]
> Something like this? Also caught a leftover print statement

That works yes. Ready to merge as soon as CI is happy!

**Comment by pepijndevos** [comment]
darn that little doc addition make ruff unhappy

**Pull request merged**